src/context/index.tsx

```diff
@@ -1,69 +1,45 @@
-import React, { FC, createContext, useState, useEffect } from "react"
-import {
-  LDClient,
-  LDFlagSet,
-  initialize as ldClientInitialize,
-} from "launchdarkly-js-client-sdk"
-import { LDUser } from "../types/ldUser"
+import React, { ReactNode, FC, createContext, useEffect, useMemo } from "react"
+import LDClient, { LDFlagSet } from "launchdarkly-js-client-sdk"
+import { LDData, LDUser } from "../types"
 
 import { camelCaseKeys } from "../lib/utils"
 
-export interface LDContext {
-  flags: LDFlagSet
-  ldClient?: LDClient
+interface Props {
+  initialLDData: LDData
+  clientId: string
+  children: ReactNode
 }
 
-export const context = createContext<LDContext>({
-  flags: {},
-  ldClient: undefined,
-})
-const { Provider, Consumer } = context
-
-interface ProviderProps {
+export interface Context {
   flags?: LDFlagSet
-  ldClientId: string
-  ldUser: LDUser
+  user?: LDUser
+  isBot: boolean
 }
 
-const initLDClient = (
-  clientSideID: string,
-  ldUser: LDUser,
-  flags: LDFlagSet
-): LDClient => {
-  return flags
-    ? ldClientInitialize(clientSideID, ldUser, { bootstrap: flags })
-    : ldClientInitialize(clientSideID, ldUser)
-}
+const LDContext = createContext<Context>()
 
-const ProviderWithState: FC<ProviderProps> = ({
-  flags,
-  ldClientId,
-  ldUser,
-  children,
-}) => {
-  const [ldClient, setLdClient] = useState<LDContext>()
+const LDProvider: FC<Props> = ({ initialLDData, clientId, children }) => {
+  // persists the data initialized server-side on the client
+  const ldData = useMemo(() => initialLDData, [])
```

Comment:
Nice usage of `useMemo` here but `initialLDData` is just an object and it is not optional so we should always get it when this provider is rendered, right? I think it is better to use it in the application, `useMemo(() => req.ldData, [])`, so we are more flexible and we can change the object if needed, wdyt?

Reply:
I think I get what you're hinting at and it's absolutely valid. `initialData` will be empty on page transitions as `getInitialProps` is called client-side and `req` is missing. The data is ignored on recurring renders and takes the memoized value. I agree that's a bit weird indeed.

I prefer to have this logic encapsulated in the pkg over imposing it on the projects though. On the initial render, the data could be changed in flight. But I believe it should not be changeable after that

Reply:
I see but TBH if we want to ensure to pass the right props, that part should be done outside of the component, but yeah..I guess it is just a preference

Reply:
I'll release it as-is and when we see an issue, I'll move it. Help me watching out for them! :)